pkg/scheduler/metrics/resources/resources.go

```diff
@@ -0,0 +1,296 @@
+			continue
+		}
+		for _, t := range []struct {
+			desc    resourceLifecycleDescriptors
+			total   v1.ResourceList
+			running v1.ResourceList
+		}{
+			{
+				desc:  podResourceDesc.requests,
+				total: reqs,
+			},
+			{
+				desc:  podResourceDesc.limits,
+				total: limits,
+			},
+		} {
+			for resourceName, val := range t.total {
+				var unitName string
+				switch resourceName {
+				case v1.ResourceCPU:
+					unitName = "cores"
+				case v1.ResourceMemory:
+					unitName = "bytes"
+				case v1.ResourceStorage:
+					unitName = "bytes"
+				case v1.ResourceEphemeralStorage:
+					unitName = "bytes"
+				default:
+					switch {
+					case isHugePageResourceName(resourceName):
+						unitName = "bytes"
+					case isAttachableVolumeResourceName(resourceName):
+						unitName = "integer"
+					}
+				}
+				var priority string
+				if p.Spec.Priority != nil {
+					priority = strconv.FormatInt(int64(*p.Spec.Priority), 10)
+				}
+				recordMetricWithUnit(ch, t.desc.total, resourceName, unitName, val, p.Namespace, p.Name, p.Spec.NodeName, p.Spec.SchedulerName, priority)
+			}
+		}
+	}
+}
+
+func recordMetricWithUnit(
+	ch chan<- metrics.Metric,
+	desc *metrics.Desc,
+	resourceName v1.ResourceName,
+	unit string,
+	val resource.Quantity,
+	namespace, name, nodeName, schedulerName, priority string,
+) {
+	if val.IsZero() {
+		return
+	}
+	ch <- metrics.NewLazyConstMetric(desc, metrics.GaugeValue,
+		val.AsApproximateFloat64(),
+		namespace, name, nodeName, schedulerName, priority, string(resourceName), unit,
+	)
+}
+
+// addResourceList adds the resources in newList to list
+func addResourceList(list, newList v1.ResourceList) {
+	for name, quantity := range newList {
+		if value, ok := list[name]; !ok {
+			list[name] = quantity.DeepCopy()
+		} else {
+			value.Add(quantity)
+			list[name] = value
+		}
+	}
+}
+
+// maxResourceList sets list to the greater of list/newList for every resource
+// either list
+func maxResourceList(list, new v1.ResourceList) {
+	for name, quantity := range new {
+		if value, ok := list[name]; !ok {
+			list[name] = quantity.DeepCopy()
+			continue
+		} else {
+			if quantity.Cmp(value) > 0 {
+				list[name] = quantity.DeepCopy()
+			}
+		}
+	}
+}
+
+// podRequestsAndLimitsByLifecycle returns a dictionary of all defined resources summed up for all
+// containers of the pod. If PodOverhead feature is enabled, pod overhead is added to the
+// total container resource requests and to the total container limits which have a
+// non-zero quantity. If the pod is assigned to a node and non-terminal, the current requests and
+// limits are returned for the pod. The caller may avoid allocations of resource lists by passing
+// a requests and limits list to the function, which will be cleared before use.
+func podRequestsAndLimitsByLifecycle(pod *v1.Pod, reuseReqs, reuseLimits v1.ResourceList) (lifecycle string, reqs, currentReqs, limits, currentLimits v1.ResourceList, terminal bool) {
```

Comment:
is "inUse" more descriptive than "current"?

Reply:
Removed code